src/util/include/util/TinyVector.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Xournal++
+ *
+ * A tiny vector stored in the stack, with a small maximal size
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <array>
+#include <stdexcept>
+
+template <class T, unsigned int N>
+class TinyVector {
+public:
+    TinyVector() = default;
+
+    using size_type = unsigned int;
+    size_type size() const { return nb; }
+    bool empty() const { return nb == 0; }
+
+    T& operator[](size_type pos) { return dataArray[pos]; }
+    const T& operator[](size_type pos) const { return dataArray[pos]; }
+    T& front() { return dataArray.front(); }
+    const T& front() const { return dataArray.front(); }
+    T& back() { return dataArray[nb - 1]; }
+    const T& back() const { return dataArray[nb - 1]; }
+
+    using const_iterator = const T*;
+    using iterator = T*;
+
+    iterator begin() { return dataArray.data(); }
+    const_iterator begin() const { return dataArray.data(); }
+    iterator end() { return dataArray.data() + nb; }
+    const_iterator end() const { return dataArray.data() + nb; }
+
+    void push_back(const T& value) {
+        if (nb < N) {
+            dataArray[nb++] = value;
+        } else {
+            throw std::length_error("TinyVector overflow");
+        }
+    }
```

Comment:
std::array accesses are tested in debug mode, so there is no reason to waste computation power here:
```suggestion
    void push_back(const T& value) {
        new (&dataArray[nb++]) T(value); 
    }
```

Reply:
Also just call emplace_back for both push_back's

Reply:
> std::array accesses are tested in debug mode, so there is no reason to waste computation power here:

I'm not sure of what you mean here. Without this test, I just get a SegFault, nothing more (with CMAKE_BUILD_TYPE = Debug).